tests/live.rs

```diff
@@ -58,26 +60,72 @@ static ALL_FILTERS: once_cell::sync::Lazy<std::sync::Mutex<adblock::lists::Filte
 
         const DEFAULT_LISTS_URL: &'static str = "https://raw.githubusercontent.com/brave/adblock-resources/master/filter_lists/default.json";
 
+        println!("Downloading list of filter lists from '{}'", DEFAULT_LISTS_URL);
         let default_lists: Vec<RemoteFilterSource> = async {
             let body = reqwest::get(DEFAULT_LISTS_URL).await.unwrap().text().await.unwrap();
             serde_json::from_str(&body).unwrap()
         }.await;
+        println!("List of filter lists has {} filter lists total", default_lists.len());
+        // ↑ Question about that:
+        // How many filter lists could a list of filter lists list if a list of filter lists could list filter lists?
+        // (asking for a friend, who may or may not be a woodchuck)
+
+        assert!(default_lists.len() > 10); // sanity check
 
         let filters_fut: Vec<_> = default_lists
             .iter()
             .map(|list| {
-                reqwest::get(&list.url)
-                    .then(|resp| resp
-                        .expect("Could not request rules")
-                        .text()
-                    ).map(move |text| (
-                            list.format,
-                            text.expect("Could not get rules as text")
-                        )
-                    )
+                println!("Starting download of filter, '{}'", list.url);
+                // 'easylist.to' is deployed via GitHub Pages. However, sometimes 'easylist.to' can
+                // take minutes to respond despite 'easylist.github.io' having no delay.
+                //
+                // In one test, the first request below ↓ took <1 second, and the second request took ~7 minutes
+                // time curl --fail -H 'Host: easylist.to' https://easylist.github.io/easylist/easylist.txt -o /dev/null
+                // time curl --fail https://easylist.to/easylist/easylist.txt -o /dev/null
+                //
+                // Toggle the `cfg` below if encountering this issue during local development.
+                // (e.g., add '--cfg override_easylist_host' to RUSTFLAGS)
+                #[cfg(any(override_easylist_host))]
+                let downloader = {
+                    let client = reqwest::Client::builder()
+                        .redirect(reqwest::redirect::Policy::none())
+                        .build()
+                        .unwrap();
+                    if list.url.starts_with("https://easylist.to") {
+                        // The use of 'http' rather than 'https' below is intentional. reqwest only
+                        // respects the host header if the target url is http, not https. (Unclear
+                        // whether that's a bug or intentional behavior.) One way to confirm that is
+                        // to send requests to http://httpbin.org/headers instead of github.io.
```

Comment:
ha, I've actually run into the exact same confusion before, for unrelated reasons: https://github.com/actix/actix-web/issues/1050

basically `Host` is forbidden for HTTP/2, the URL ends up being the source of truth.